protocol/x/vault/keeper/shares.go

```diff
@@ -0,0 +1,34 @@
+package keeper
+
+import (
+	"cosmossdk.io/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dydxprotocol/v4-chain/protocol/x/vault/types"
+)
+
+// GetTotalShares gets TotalShares for a vault.
+func (k Keeper) GetTotalShares(
+	ctx sdk.Context,
+	vaultId types.VaultId,
+) (val types.NumShares, exists bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.TotalSharesKeyPrefix))
```

Comment:
This is finding the existing keyvalue store that is associated with this keeper? Why do we use this instead of just storing this KV store directly in the keeper in `x/vault/keeper.go`?

Reply:
We usually create a separate store (within `KVStore`) for each "functionality"